lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -1578,15 +1578,19 @@ void ByteCodeGenerator::EmitScopeObjectInit(FuncInfo *funcInfo)
                     Assert(sym->GetScopeSlot() != Js::Constants::NoProperty && sym->GetScopeSlot() > slot);
                     propIds->elements[slot] = Js::Constants::NoProperty;
                 }
-                slot++;
             }
+            else
```

Comment:
It's a bit unclear to me here from just reading the code what this else is meant to handle. Adding something to restrict to the pattern case or assert that would helpful here.

---
In reply to: [58244592](https://github.com/Microsoft/ChakraCore/pull/715#discussion_r58244592) [](ancestors = 58244592)